test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright(c) 2019-2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import os
+import pytest
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CacheModeTrait, CacheLineSize, CleaningPolicy, \
+    FlushParametersAcp
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils import os_utils
+from test_utils.filesystem.file import File
+from test_utils.os_utils import DropCachesMode
+from test_utils.size import Size, Unit
+from test_utils.time import Time
+from tests.lazy_writes.recovery.recovery_tests_methods import power_cycle_dut
+
+test_file_size = Size(300, Unit.MebiByte)
+mount_point = "/mnt"
+filename = "fio_test_file"
+pattern = "0xabcd"
+other_pattern = "0x0000"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.parametrize("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_plugin("power_control")
+def test_recovery_all_options(cache_mode, cache_line_size, cleaning_policy, filesystem):
+    """
+        title: Test for recovery after reset with various cache options.
+        description: Verify that unflushed data can be safely recovered after reset.
+        pass_criteria:
+          - CAS recovers successfully after reboot
+          - No data corruption
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(200, Unit.MebiByte)])
+        core_disk.create_partitions([Size(2000, Unit.MebiByte)] * 2)
```

Comment:
Where do You use the second partition?

Reply:
second partition is created to make sure that after reboot CAS loads with use of correct partition - that's why I use symlinks, because sometimes after reboot device name is changed but symlink stays the same.

Reply:
Thanks for explanation.